--- credentials-control-center-0.1.5/libaccount-plugin/oauth-plugin.c.orig	2013-03-25 09:57:42.000000000 +0100
+++ credentials-control-center-0.1.5/libaccount-plugin/oauth-plugin.c	2019-04-22 20:56:50.912695351 +0200
@@ -41,6 +41,7 @@
 #include <libsignon-glib/signon-auth-session.h>
 #include <libsignon-glib/signon-errors.h>
 #include <libsignon-glib/signon-identity.h>
+#include <libsignon-glib/signon-security-context.h>
 
 enum
 {
@@ -184,7 +185,7 @@
         !priv->deleting_identity)
     {
         priv->deleting_identity = TRUE;
-        signon_identity_remove (priv->identity, identity_removed_cb, self);
+        signon_identity_remove (priv->identity, NULL, identity_removed_cb, self);
     }
 
     finish_if_ready (self);
@@ -342,7 +343,7 @@
 {
     ApOAuthPluginPrivate *priv = self->priv;
 
-    signon_identity_query_info (priv->identity, query_info_cb, self);
+    signon_identity_query_info (priv->identity, NULL, query_info_cb, self);
 }
 
 static void
@@ -504,7 +505,7 @@
         return;
     }
     priv->authenticating = TRUE;
-    signon_auth_session_process_async (priv->auth_session, session_data,
+    signon_auth_session_process (priv->auth_session, session_data,
                                        get_mechanism (priv),
                                        priv->cancellable,
                                        auth_session_process_cb,
@@ -550,10 +551,11 @@
 {
     SignonIdentityInfo *info;
     AgProvider *provider;
-    const gchar *acl_all[] = { "*", NULL };
     const gchar *username;
     const gchar *secret;
+    GList *acl_list = NULL;
 
+    acl_list = g_list_append (acl_list, signon_security_context_new_from_values ("*", NULL));
     username = ap_plugin_get_username ((ApPlugin *)self);
     secret = ap_plugin_get_password ((ApPlugin *)self);
     if (secret == NULL) secret = "";
@@ -566,10 +568,11 @@
     if (username != NULL)
         signon_identity_info_set_username (info, username);
     signon_identity_info_set_secret (info, secret, TRUE);
-    signon_identity_info_set_access_control_list (info, acl_all);
+    signon_identity_info_set_access_control_list (info, acl_list);
+    g_list_free_full (acl_list, (GDestroyNotify)signon_security_context_free);
 
     self->priv->identity = signon_identity_new ();
-    signon_identity_store_credentials_with_info (self->priv->identity, info,
+    signon_identity_store_info (self->priv->identity, info, NULL,
                                                  identity_store_cb, self);
     signon_identity_info_free (info);
 }
--- credentials-control-center-0.1.5/libaccount-plugin/plugin.c.orig	2013-03-25 09:57:42.000000000 +0100
+++ credentials-control-center-0.1.5/libaccount-plugin/plugin.c	2019-04-22 21:17:13.259406664 +0200
@@ -285,6 +285,7 @@
         if (identity != NULL) {
             deleting_identity = TRUE;
             signon_identity_remove (identity,
+                                    NULL,
                                     identity_removed_cb,
                                     result);
         }
